pkg/model/sshkey.go

```diff
@@ -41,6 +41,8 @@ func (b *SSHKeyModelBuilder) Build(c *fi.ModelBuilderContext) error {
 	t := &awstasks.SSHKey{
 		Name:      s(name),
 		Lifecycle: b.Lifecycle,
+		Tags:      b.CloudTags(b.ClusterName(), false),
+		Shared:    b.Cluster.Spec.SSHKeyName != nil && *b.Cluster.Spec.SSHKeyName != "",
```

Comment:
May be better:
```suggestion
		Shared:    fi.StringValue(b.Cluster.Spec.SSHKeyName) != "",
```
https://github.com/kubernetes/kops/blob/1c37a328e1a76563f3fb2a41b9353675653650b4/upup/pkg/fi/values.go#L25-L30